### PR TITLE
Modernize OpenDdl and StanfordImporter

### PR DESCRIPTION
**Pull request by alicemargatroid**
Real small PR, replaces push_back with emplace_back and adds some qualifiers to improve performance

### CODE REVIEW

File: src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h [hunk 1]
```diff
@@ -428,7 +428,7 @@ class MAGNUM_TRADE_OPENGEXIMPORTER_EXPORT Document {
 
         MAGNUM_TRADE_OPENGEXIMPORTER_LOCAL std::size_t dereference(std::size_t originatingStructure, Containers::ArrayView<const char> reference) const;
 
-        MAGNUM_TRADE_OPENGEXIMPORTER_LOCAL bool validateLevel(std::optional<Structure> first, Containers::ArrayView<const std::pair<Int, std::pair<Int, Int>>> allowedStructures, Containers::ArrayView<const Validation::Structure> structures, std::vector<Int>& counts) const;
+        MAGNUM_TRADE_OPENGEXIMPORTER_LOCAL bool validateLevel(const std::optional<Structure>& first, Containers::ArrayView<const std::pair<Int, std::pair<Int, Int>>> allowedStructures, Containers::ArrayView<const Validation::Structure> structures, std::vector<Int>& counts) const;
```
Comment (mosra) [inline src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h]: Hmm. The `Structure` is just two pointers in a POD, so I thought it is completely okay to pass it via value. But the `std::optional` probably adds some nontrivial boilerplate around so it's not as free to copy anymore.

**Comment by mosra** [comment]
Merged, thank you!

File: src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h (hunk 1 above)
Comment (Squareys) [inline src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h]: *POD*? Plain old ...? 😅

Comment (LB--) [inline src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h]: @Squareys Plain Old Data, unless you're suggesting something else?

Comment (mosra) [inline src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h]: Data. :D

Comment (alicemargatroid) [inline src/MagnumPlugins/OpenGexImporter/OpenDdl/Document.h]: The current term would be "Standard Layout"; we're working in C++11 I think.
But std::optional is not standard layout, I believe.

**Pull request merged**